src/components/organisms/edit-denominations-modal/index.tsx

```diff
@@ -0,0 +1,160 @@
+import React from "react"
+import { v4 as uuidv4 } from "uuid"
+import Button from "../../fundamentals/button"
+import PlusIcon from "../../fundamentals/icons/plus-icon"
+import TrashIcon from "../../fundamentals/icons/trash-icon"
+import InfoTooltip from "../../molecules/info-tooltip"
+import Modal from "../../molecules/modal"
+import CurrencyInput from "../../organisms/currency-input"
+
+export type PriceType = {
+  currency_code: string
+  amount: number
+  id?: string
+}
+
+type EditDenominationsModalProps = {
+  defaultDenominations: PriceType[]
+  handleClose: () => void
+  onSubmit: (denominations: PriceType[]) => void
+  defaultNewAmount?: number
+  defaultNewCurrencyCode?: string
+  currencyCodes?: string[]
+}
+
+const augmentWithId = (obj) => ({ ...obj, id: obj.id ? obj.id : uuidv4() })
+
+const augmentWithIds = (list) => {
+  return list.map(augmentWithId)
+}
+
+const EditDenominationsModal = ({
+  defaultDenominations = [],
+  onSubmit,
+  handleClose,
+  currencyCodes,
+  defaultNewAmount = 1000,
+  defaultNewCurrencyCode = "USD",
+}: EditDenominationsModalProps) => {
+  const [denominations, setDenominations] = React.useState(
+    augmentWithIds(defaultDenominations)
+  )
+
+  const onAmountChange = (index) => {
+    return (amount) => {
+      const newDenominations = denominations.slice()
+      newDenominations[index] = { ...newDenominations[index], amount }
+      setDenominations(newDenominations)
+    }
+  }
+
+  const onCurrencyChange = (index) => {
+    return (currencyCode) => {
+      const newDenominations = denominations.slice()
+      newDenominations[index] = {
+        ...newDenominations[index],
+        currency_code: currencyCode,
+      }
+      setDenominations(newDenominations)
+    }
+  }
+
+  const onClickDelete = (index) => {
+    return () => {
+      const newDenominations = denominations.slice()
+      newDenominations.splice(index, 1)
+      setDenominations(newDenominations)
+    }
+  }
+
+  const appendDenomination = (newDenomination) => {
+    setDenominations([...denominations, augmentWithId(newDenomination)])
+  }
```

Comment:
nit: Think it would be cool to add two rules to this operation:
1. When appending a new denomination use a currency code that is not already present among the other denominations
2. Add a limit on number of denominations that correlates to the number of different currencies in store

![image](https://user-images.githubusercontent.com/59018053/151957351-2ed53fa5-d501-4f60-81d4-ca2e9b081ec6.png)

Reply:
E.g. if I have DKK, USD, and EUR as store currencies, the above image should disable "+ Add a price" when there are three rows and row 3 should be EUR instead of USD

Reply:
@olivermrbl I pushed a fix for the above. I decided not to interact with our store explicitly from the modal and instead, act entirely on the `currencyCodes` prop. This allows the EditDenominationsModal to stay as "dumb" as possible. You could achieve what you mentioned above by doing something like this from any component that consumes the modal: 
```jsx
const Component = () => {
  const { store } = useAdminStore()
  
  return (
    <EditDenominationsModal 
        currencyCodes={store.currencies.map(c => c.code)} 
        onSubmit={console.log} 
        handleClose={handleClose}
    />
  )
}
```

What do you think about this approach? Do you prefer keeping the modal as "pure" as possible, or would you rather we call `useAdminStore()` from the modal? 

Also, feel free to use the storybook link deploy link to play around with this :smile:

Reply:
I really like this @zakariaelas! Agree, that we should strive to keep modals pure - great work 👍